auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```diff
@@ -318,82 +323,83 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
         parameters: Map<String, String>,
         callback: Callback<Credentials, CredentialsManagerException>
     ) {
-        val encryptedEncoded = storage.retrieveString(KEY_CREDENTIALS)
-        val encrypted = Base64.decode(encryptedEncoded, Base64.DEFAULT)
-        val json: String
-        try {
-            json = String(crypto.decrypt(encrypted))
-        } catch (e: IncompatibleDeviceException) {
-            callback.onFailure(
-                CredentialsManagerException(
-                    String.format(
-                        "This device is not compatible with the %s class.",
-                        SecureCredentialsManager::class.java.simpleName
-                    ), e
+        serialExecutor.execute {
+            val encryptedEncoded = storage.retrieveString(KEY_CREDENTIALS)
+            val encrypted = Base64.decode(encryptedEncoded, Base64.DEFAULT)
+            val json: String
+            try {
+                json = String(crypto.decrypt(encrypted))
+            } catch (e: IncompatibleDeviceException) {
+                callback.onFailure(
+                    CredentialsManagerException(
+                        String.format(
+                            "This device is not compatible with the %s class.",
+                            SecureCredentialsManager::class.java.simpleName
+                        ), e
+                    )
                 )
-            )
-            decryptCallback = null
-            return
-        } catch (e: CryptoException) {
-            //If keys were invalidated, existing credentials will not be recoverable.
-            clearCredentials()
-            callback.onFailure(
-                CredentialsManagerException(
-                    "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
-                            "Any previously stored content is now lost. Please, try saving the credentials again.",
-                    e
+                decryptCallback = null
+                return@execute
+            } catch (e: CryptoException) {
+                //If keys were invalidated, existing credentials will not be recoverable.
+                clearCredentials()
+                callback.onFailure(
+                    CredentialsManagerException(
+                        "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
+                                "Any previously stored content is now lost. Please, try saving the credentials again.",
```

Comment:
I know this is just copied from the old diff but the grammar is incorrect here, needs the comma removed.

```suggestion
                                "Any previously stored content is now lost. Please try saving the credentials again.",
```